common/lib/container-definitions/src/runtime.ts

```diff
@@ -166,6 +169,24 @@ export interface IRuntimeFactory extends IProvideRuntimeFactory {
     /**
      * Instantiates a new IRuntime for the given IContainerContext to proxy to
      * This is the main entry point to the Container's business logic
+     *
+     * @deprecated Use getRuntime
      */
     instantiateRuntime(context: IContainerContext): Promise<IRuntime>;
+
+    getRuntime(context: IContainerContext, existing?: boolean): Promise<IRuntime>;
+
+    /**
+     * Instantiates a new IRuntime for the given IContainerContext to proxy to
+     * Performs the appropriate initialization for a new runtime
+     *
+     * This is the main entry point to the Container's business logic
+     */
+    instantiateFirstTime(context: IRuntime): Promise<void>;
```

Comment:
I'm thinking that maybe we don't need `instantiateFirstTime` and `instantiateFromExisting` in the interface and only keep them in the helper class. This way, the only change to this would be adding `getRuntime`.